### PR TITLE
Improve retrieval of keyword values from FITS files.

### PR DESCRIPTION
**Pull request by judith-ipac**

### CODE REVIEW

**Review by smonkewitz — APPROVED** [review]
The CCfits code dealing with Keyword::value specializations is pretty hard to follow, but this looks like it's about the best we can do. Their docs for the templated `value` method are incorrect (from `CCfits/Keyword.h`):

> Allowed T types: CCfits stores keyword values of type U in a templated subclass of Keyword, KeyData<U>. Normally U is set when reading the Keyword in from the file, and is limited to types int, double, string, bool, and complex<float>.

I guess `bool` should be added to the list in the docs above...

Regarding the fallback - I am a bit worried that something like a `Tlonglong` value would use the implicit conversion to `int` code path and get truncated as a result, but it's not clear that CCfits will ever use that type for keywords in headers it is reading. I think it would be OK to deal with further issues like this one as they arise.

**Comment by judith-ipac** [comment]
I haven't looked into creating FITS files from scratch so haven't experimented with Tlonglong.  I'll merge to master but will be prepared to revisit if need be.
